src/conn/conn_reconfig.c

```diff
@@ -0,0 +1,212 @@
+/*-
+ * Copyright (c) 2014-2017 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_conn_compat_config --
+ *	Configure compatibility version.
+ */
+int
+__wt_conn_compat_config(WT_SESSION_IMPL *session, const char **cfg)
+{
+	WT_CONFIG_ITEM cval;
+	WT_CONNECTION_IMPL *conn;
+	uint16_t patch;
+	bool txn_active;
+
+	conn = S2C(session);
+	WT_RET(__wt_config_gets(session, cfg,
+	    "compatibility.release", &cval));
+	if (cval.len == 0) {
+		conn->compat_major = WIREDTIGER_VERSION_MAJOR;
+		conn->compat_minor = WIREDTIGER_VERSION_MINOR;
+		return (0);
+	}
+
+	/*
+	 * Accept either a major.minor release string or a
+	 * major.minor.patch release string.  We ignore the patch
+	 * value, but allow it in the string.
+	 */
+	if (sscanf(cval.str, "%" SCNu16 ".%" SCNu16,
+	    &conn->compat_major, &conn->compat_minor) != 2 &&
+	    sscanf(cval.str, "%" SCNu16 ".%" SCNu16 ".%" SCNu16,
+	    &conn->compat_major, &conn->compat_minor, &patch) != 3)
+		WT_RET_MSG(session,
+		    EINVAL, "illegal compatibility release");
+	if (conn->compat_major > WIREDTIGER_VERSION_MAJOR)
+		WT_RET_MSG(session, EINVAL, "unknown major version");
+	if (conn->compat_major == WIREDTIGER_VERSION_MAJOR &&
+	    conn->compat_minor > WIREDTIGER_VERSION_MINOR)
+		WT_RET_MSG(session,
+		    EINVAL, "illegal compatibility version");
+
+	/*
+	 * We're doing an upgrade or downgrade, check whether transactions are
+	 * active.  Wait for any in-progress checkpoints to complete.
+	 */
```

Comment:
This comment is inconsistent with the code here.  There is no waiting for checkpoints in this code.